yoga/Yoga.c

```diff
@@ -362,6 +363,22 @@ YGNodeRef YGNodeNew(void) {
   return YGNodeNewWithConfig(&gYGConfigDefaults);
 }
 
+YGNodeRef YGNodeClone(const YGNodeRef oldNode) {
+  const YGNodeRef node = gYGMalloc(sizeof(YGNode));
+  YGAssertWithConfig(oldNode->config, node != NULL, "Could not allocate memory for node");
+  gNodeInstanceCount++;
+
+  memcpy(node, oldNode, sizeof(YGNode));
+  if (YGNodeGetChildCount(node) == 0) {
+    // If the node list count is zero, we reset it to NULL to indicate that we shouldn't
+    // reuse the children list when we mutate it. If it's non-empty we use the content of the
+    // list to determine whether to fork it.
+    node->children = NULL;
+  }
```

Comment:
I am concerned about reusing the node list itself since `node->children` is actually owned by `node`. So freeing one of the parent nodes can free the node list itself.

An alternative would be to clone the list itself. However, that's unfortunate for the case when you're just going to clear the list right after. Although even if you do, you're probably just going to fill it up with roughly as many children.

I could also add a CloneAndRemoveAllChildren API that clears the list. Such as if you know you're going to clone the children anyway.

Reply:
Fixed in https://github.com/facebook/yoga/pull/636/commits/822bd491f06fb95e531eb2e7c5ce276eaa636571